bookshelf/6-gce/src/main/java/com/example/getstarted/objects/Book.java

```diff
@@ -0,0 +1,180 @@
+/* Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.getstarted.objects;
+
+// [START example]
+public class Book {
+// [START book]
+  private String title;
+  private String author;
+  private String createdBy;
+  private String createdById;
+  private String publishedDate;
+  private String description;
+  private Long id;
+  private String imageUrl;
+// [END book]
+// [START keys]
+  public static final String AUTHOR = "author";
+  public static final String CREATED_BY = "createdBy";
+  public static final String CREATED_BY_ID = "createdById";
+  public static final String DESCRIPTION = "description";
+  public static final String ID = "id";
+  public static final String PUBLISHED_DATE = "publishedDate";
+  public static final String TITLE = "title";
+  public static final String IMAGE_URL = "imageUrl";
+// [END keys]
+// [START constructor]
+  // We use a Builder pattern here to simplify and standardize construction of Book objects.
+  private Book(Builder builder) {
+    this.title = builder.title;
+    this.author = builder.author;
+    this.createdBy = builder.createdBy;
+    this.createdById = builder.createdById;
+    this.publishedDate = builder.publishedDate;
+    this.description = builder.description;
+    this.id = builder.id;
+    this.imageUrl = builder.imageUrl;
+  }
+// [END constructor]
+// [START builder]
+  public static class Builder {
```

Comment:
Eamonn didn't think laying AutoValue on users at this point was a good idea.  I was good w/ the idea.